utils/python/CIME/XML/batch.py

```diff
@@ -56,12 +56,16 @@ def set_batch_system(self, batch_system, machine=None):
         """
         machine = machine if machine is not None else self.machine
         if self.batch_system != batch_system or self.batch_system_node is None:
-            self.batch_system_node = self.get_optional_node("batch_system", {"type" : batch_system})
-            expect(self.batch_system_node is not None, "No batch system '%s' found" % batch_system)
+            nodes = self.get_nodes("batch_system",{"type" : batch_system})
```

Comment:
there can be more than one batch_system node in the file with type batch_system but only one for each machine